### ISSUE TITLE
Feature: Workplace Hazards

### ISSUE THREAD

**Issue by JR1811** [LopyLuna/Create-Calamity#14]
## Content

Adds Workplace Hazards to the game.

Examples are:

### New types of damage

- [x] Damage when caught between cogs (damage depending on SU?)
- [x] Wall belts fling entities around
- [x] Mechanical presses damages entities below

### Pipe pressure system

- [ ] Pressure in breakable pipes with jet of leaking fluid damaging entities
- [ ] safety valves for boilers to lower pressure in pipes

## Contributors

- Dev - **rdh**
- Dev - **Pouffy**
- Dev - **Sascha_T**
- Dev - *[Add more information]*
- Artist - *[Add more information]*

**Comment by rhysdh540** [comment]
Oh no

**Comment by JR1811** [comment]
Its official. Change the description however u want

**Comment by Sascha-T** [comment]
> Oh no

Oh **YES**


~~wild feature with 3 people vying to dev it~~

**Comment by Sascha-T** [comment]
Should the press crushing effect be applied on `tryProcessInWorld`/`tryProcessOnBelt`?
Insta-kill or not, and if not, what form should it take? Suffocation during a certain height range of the press animation?
Can we clarify?

**Comment by Sascha-T** [comment]
![death](https://github.com/LopyLuna/Create-Calamity/assets/16324266/4dc2369d-d2ac-40f3-a1f0-71464453bf8b)

**Comment by JR1811** [comment]
Would be nice if it would already press whenever it detects an Entity.

**Comment by Sascha-T** [comment]
> Would be nice if it would already press whenever it detects an Entity.
🫡

![code in question](https://github.com/LopyLuna/Create-Calamity/assets/16324266/6008111b-9304-4927-b35a-dfc1932925e9)
![reaction](https://github.com/LopyLuna/Create-Calamity/assets/16324266/dd059266-8f3b-4c8c-9e95-62329298db8a)

it is a bit aggressive and will crush you if you are even *slightly* in range, we may choose to reduce damage for people not completely inside or reduce the AABB on the entity detector

but first, we should obviously make the press capable of forcing players into a crouching/crawling position, right? :)

**Comment by pouffy** [comment]
With the press damage source, make it damage helmets specifically. also the speed should relate to the damage.

**Comment by Sascha-T** [comment]
> With the press damage source, make it damage helmets specifically. also the speed should relate to the damage.

Durability damage relative to speed, damage to player relative to speed, did I get that right?
And we could also have it play a "THUNK" or something and skip past the cycle (return to base)

**Comment by JR1811** [comment]
wouldnt SU for the entity damage make more sense tho?

**Comment by Sascha-T** [comment]
> wouldnt SU for the entity damage make more sense tho?

true + send a formula

**Comment by pouffy** [comment]
> wouldnt SU for the entity damage make more sense tho?

Yes but if a mechanical press slams down on your head at 256rpm that would likely deal more damage than it slowly pressing down at 16rpm

**Comment by Sascha-T** [comment]
> > wouldnt SU for the entity damage make more sense tho?
> 
> Yes but if a mechanical press slams down on your head at 256rpm that would likely deal more damage than it slowly pressing down at 16rpm

if it had enough pressure to kill you, the speed doesnt matter is the argument ig?

**Comment by pouffy** [comment]
I feel like if it's getting crushed by a fast moving component the damage should be instant and large, but if is is slow the damage should be more spread out (kinda like crushing wheels). The pressure would still damage you but the speed makes the death faster.

**Comment by Sascha-T** [comment]
I committed my additions for the day to [the feat/workplace-hazards branch.](https://github.com/LopyLuna/Create-Calamity/tree/feat/workplace-hazards)

I'll get back to discussing and considering that spread-out damage mechanic with you tommorow (or you can just mess up everything, write it yourself and push it in overnight while I'm asleep :p, you choose)

**Comment by Sascha-T** [comment]
flingy belts exist now as of 0d860fe

**Comment by Sascha-T** [comment]
cog crushing is so incredibly scuffed :(

**Comment by Sascha-T** [comment]
scuffed but done

**Comment by CodedKyoko** [comment]
Players should be able to fall down chutes
+ extra fall damage if the chute uses one of those horizontal connectoin chutes

**Comment by JR1811** [comment]
> Players should be able to fall down chutes

How are they supposed to be handled if there are the diagonal chutes?

**Comment by CodedKyoko** [comment]
@JR1811 

> > Players should be able to fall down chutes
> 
> How are they supposed to be handled if there are the diagonal chutes?

as I said, extra fall damage, but they continue falling till they hit the end the the chute

**Comment by JR1811** [comment]
> as I said, extra fall damage, but they continue falling till they hit the end the the chute

I rly feel like they should get damaged instantly when they bang their head against the metal and keep their normal fall damage handling ngl xD
